### DISCUSSION TITLE
ERROR: failed to solve: the Dockerfile cannot be empty

### DISCUSSION THREAD

**Discussion by ckakarladevops** [learndevopsonline/learndevopsonline#1873] [Q&A]
Hi Team,
I am unable to create own image using docker file, tried so many times, but not possible. Also i found that, code is not pushing from local machine to git hub repository. All the commands are executed successfully, but still it's not showing in Git hub. All the screenshots attached. Could you please check and guide me how to proceed.

![Commands executed in local machine](https://github.com/learndevopsonline/learndevopsonline/assets/123289728/bfcdcedc-fdd2-43da-b90a-ec27d69ccf8c)
![Executed Code in VC](https://github.com/learndevopsonline/learndevopsonline/assets/123289728/545951b2-8437-45cf-8186-006c3515a9aa)
![git hub repository](https://github.com/learndevopsonline/learndevopsonline/assets/123289728/d8eb0099-67a9-4c3d-909e-00ca4948e9d0)

[Docker File Error.txt](https://github.com/learndevopsonline/learndevopsonline/files/14795906/Docker.File.Error.txt)

**Comment by chaithanya1812** [comment]
## Before git push, you need to follow below steps
git add .
git commit -m  "commit-message"

**Reply by ckakarladevops** [reply]
Hi chaitanya,
I have already run before that, but the same.

**Comment by chaithanya1812** [comment]
Join the lab  query session

**Reply by ckakarladevops** [reply]
Hi Chaitanya... Can you provide me the lab query session link and timings as well.

**Comment by chaithanya1812** [comment]
Maybe I can't find exact solution until you join lab session
## check once this might be one of the issue, you doing changes in visual studio & pushing your code using cli ( maybe  changes are not saved to a files)  
![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/85cf6be9-97d3-4fd8-9c32-4120abf0c021)

If it's not working, -> run this command ( git status ) & post the error

**Reply by ckakarladevops** [reply]
![git status report](https://github.com/learndevopsonline/learndevopsonline/assets/123289728/72cf15b8-a2ca-41c3-9fbc-08e8e92b45df)

**Comment by chaithanya1812** [comment]
Open the file on cli, using vi command & add your code to a file.
Then run git status & post it..

**Reply by ckakarladevops** [reply]
Hi Chaitanya, Please share me lab query session details. i will join and share you the error.  I really strucked here and not moving further sessions.

**Reply by chaithanya1812** [reply]
https://learndevopsonline.my.canva.site/lab-help
